### ISSUE TITLE
sphinx.main(argv) ignores argv

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1611]
The `sphinx.main()` function, as implemented by `cmdline.main()`, accepts an "argv" argument. This argument however is not passed to the option parser, which then uses `sys.argv` as the default.

This is not an issue for the main sphinx driver because it was passing sys.argv to sphinx.main anyway. However, when you want to call sphinx.main programmatically from a Python program that also does other stuff, it break down.

The patch below fixes the issue:

```
diff -r 97de33602535 sphinx/cmdline.py
--- a/sphinx/cmdline.py Thu Oct 30 10:13:54 2014 +0900
+++ b/sphinx/cmdline.py Sat Nov 01 21:55:56 2014 -0400
@@ -126,7 +126,7 @@

     # parse options
     try:
-        opts, args = parser.parse_args()
+        opts, args = parser.parse_args(argv[1:])
     except SystemExit as err:
         return err.code

```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1611
- Originally reported by: Geert Jansen
- Originally created at: 2014-11-02T03:04:01.798

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-11-04 07:51:53+00:00_

Thanks for the report!

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-11-04 07:52:03+00:00_

Closes #1611: actually use "argv" argument again in cmdline.main.

→ <<cset fdeaff759a389a449b78376c0985df406b80bee9>>
